var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -145,373 +142,72 @@ class IntelParallelStudio(IntelPackage):
-                self.spec.mpicxx = bindir.mpiicpc
-                self.spec.mpifc  = bindir.mpiifort
-                self.spec.mpif77 = bindir.mpiifort
-            else:
-                self.spec.mpicc  = bindir.mpicc
-                self.spec.mpicxx = bindir.mpicxx
-                self.spec.mpifc  = bindir.mpif90
-                self.spec.mpif77 = bindir.mpif77
-
-    def setup_environment(self, spack_env, run_env):
-        """Adds environment variables to the generated module file.
-
-        These environment variables come from running:
-
-        .. code-block:: console
-
-           $ source parallel_studio_xe_2017/bin/psxevars.sh intel64
-        """
-        # NOTE: Spack runs setup_environment twice, once pre-build to set up
-        # the build environment, and once post-installation to determine
-        # the environment variables needed at run-time to add to the module
-        # file. The script we need to source is only present post-installation,
-        # so check for its existence before sourcing.
-        # TODO: At some point we should split setup_environment into
-        # setup_build_environment and setup_run_environment to get around
-        # this problem.
-        psxevars = glob.glob(join_path(
-            self.prefix, 'parallel_studio*', 'bin', 'psxevars.sh'))
-
-        if psxevars:
-            run_env.extend(EnvironmentModifications.from_sourcing_file(
-                psxevars[0], 'intel64'))
+# TODO: I don't think this still applies as of 2017.3 and later:
+#
+# $ grep "SCRIPTPATH=" parallel_studio_xe_201?.*/psxevars.sh
+# par..._xe_2016.0.047/psxevars.sh:SCRIPTPATH=/opt/intel
+# par..._xe_2016.1.056/psxevars.sh:SCRIPTPATH=/opt/intel
+# par..._xe_2016.2.062/psxevars.sh:SCRIPTPATH=/opt/intel
+# par..._xe_2016.3.067/psxevars.sh:SCRIPTPATH=/opt/intel
+# par..._xe_2016.4.072/psxevars.sh:SCRIPTPATH=/opt/intel
+# par..._xe_2017.0.035/psxevars.sh:SCRIPTPATH=/opt/intel
+# par..._xe_2017.3.053/psxevars.sh:SCRIPTPATH=/opt/intel/par..._xe_2017.3.053
+# par..._xe_2017.4.056/psxevars.sh:SCRIPTPATH=/opt/intel/par..._xe_2017.4.056
+# par..._xe_2017.5.061/psxevars.sh:SCRIPTPATH=/opt/intel/par..._xe_2017.5.061
+# par..._xe_2017.6.064/psxevars.sh:SCRIPTPATH=/opt/intel/par..._xe_2017.6.064
+# par..._xe_2018.1.038/psxevars.sh:SCRIPTPATH=/opt/intel/par..._xe_2018.1.038
+#
+# (I did not have .1 and .2 installed)
+
+        f = self.file_to_source
+        if self.version > Version('2016') and self.version < Version('2017.3'):
+            filter_file('^SCRIPTPATH=.*',
+                        'SCRIPTPATH={0}'.format(self.prefix),
+                        f)
+        # Don't bother with the csh version.
+
+    def setup_dependent_environment(self, *args):
+        # CAUTION - DUP code in:
+        #   ../intel-mpi/package.py
+        #   ../intel-parallel-studio/package.py
+        # Tail-call the actual implementation, conveying client's compilers.
+        return self.mpi_setup_dependent_environment(
```

Comment:
This looks like it needs an `if '+mpi' in self.spec:`

Reply:
Addressed.